### PR TITLE
[JENKINS-61300] Update search box size on page resize, max width

### PR DESCRIPTION
**Pull request by zbynek**
To reproduce:
* open Jenkins in a small browser window (small enough for search to be hidden)
* make the window bigger
* type in  long text

Expected:
1) after resize the search box is reasonably big
2) when typing, autocompletion size follows search box size

Actual:
1) after resize there is a 60px wide search box
image

2) when typing, on key up the autocomplete size syncs with box, but resizes back when autocomplete results are fetched

This PR fixes 1) by adding a resize listener and 2) by applying minWidth instead of width. Also removes extra div that was used for minWidth computation and uses CSS min-width instead.

### Proposed changelog entries

* Entry 1: Update size of the search box properly when screen is resized.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [n/a] JIRA issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [n/a] Appropriate autotests or explanation to why this change has no tests
- [n/a] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@fqueiruga 

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a JIRA issue should exist and be labeled as `lts-candidate`

### CODE REVIEW

**Review by timja — APPROVED** [review]

File: war/src/main/webapp/scripts/hudson-behavior.js
Comment (fqueiruga) [inline war/src/main/webapp/scripts/hudson-behavior.js]: In general, I'm very wary of resize listeners, as they can have devastating effects on performance. Have you explored other options?

If a resize listener is the way to go, I'd say we need to throttle it. One option could be using `window.requestAnimationFrame`. That would sort of ensure we can keep 60fps. It's present on all  the supported browsers, and we could just not set this listener on <IE9, as it's not essential.

We can also set the listeners just on desktop screens. We can use `window.matchMedia` for that (https://developer.mozilla.org/en-US/docs/Web/API/MediaQueryList/addListener).

File: war/src/main/webapp/scripts/hudson-behavior.js
Comment (fqueiruga) [inline war/src/main/webapp/scripts/hudson-behavior.js]: I'd suggest we only run the contents of the `updatePos` function only when the window width changes. We should avoid running this function on vertical resizes.

**Review by fqueiruga — CHANGES_REQUESTED** [review]
Love this PR! I just have a few comments.

I also think this should be reported as a bug on the issue tracker. That way we can backport this to the LTS if the header makes it into the cut.

File: war/src/main/webapp/scripts/hudson-behavior.js
Comment (fqueiruga) [reply war/src/main/webapp/scripts/hudson-behavior.js]: I just tried this and I think we can skip the listener altogether if we remove the first invocation of `updatePos()`. Wonder if that can have any consequence. Having min-width already does the trick.

File: war/src/main/webapp/scripts/hudson-behavior.js
Comment (fqueiruga) [reply war/src/main/webapp/scripts/hudson-behavior.js]: Disregard the above comment, that would only be the case if it runs on the window resize event, not on keyup

**Review by fqueiruga — APPROVED** [review]
I love this change, specially having a maximum width for the search bar.

If this was thoroughly tested and the rest of the community feels OK with it I'd argue it can be an lts-candidate. An argument against that can be that it does not fix a major bug though.

**Comment by zbynek** [comment]
The original bug is way older than the new header, so I see no rush putting this in LTS. But if you think a backport is needed, I can create a Jira issue.

**Comment by zbynek** [comment]
In case a Jira issue is expected, I created one: https://issues.jenkins-ci.org/browse/JENKINS-61300

**Comment by timja** [comment]
> In case a Jira issue is expected, I created one: [issues.jenkins-ci.org/browse/JENKINS-61300](https://issues.jenkins-ci.org/browse/JENKINS-61300)

only needed here if you think it should be backported, this is just waiting for another review or two

**Review by oleg-nenashev — APPROVED** [review]
Looks good to me. Since @fqueiruga approved the pull request, I plan to merge it in 24 hours if there is no negative feedback

**Pull request merged**
